content/docs/pipelines/sdk/static-type-checking.md

```diff
@@ -0,0 +1,144 @@
++++
+title = "DSL Static Type Checking"
+description = "Statically check the component I/O types"
+weight = 6
++++
+
+## Overview
+A pipeline is a workflow consisting of [components](/docs/pipelines/sdk/build-component#overview-of-pipelines-and-components) and each
+component contains inputs and outputs. The DSL compiler supports static type checking to ensure the type consistency among the component
+I/Os within the same pipeline. Static type checking helps users to identify component I/O inconsistencies without running the pipeline. 
+It also shortens the development cycles by catching the errors early. 
+This feature is especially useful in two cases: 
+* When the pipeline is huge and manually checking the types is infeasible; 
+* When some components are shared ones and the type information is not immediately available to the pipeline authors.
+
+## Type System  
+In Kubeflow pipeline, a type is defined as a type name with an [OpenAPI Schema](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md)
+property, which defines the input parameter schema. *Warning*( the pipeline system 
+currently does not check the input value against the shcema when users submit a pipeline run. However, this feature will come in the near 
+future). There is a set of [core types](https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/dsl/_types.py) defined in the 
+pipeline SDK and users might use these core types or define their custom types. 
+
+In the component YAML, types are specified as a string or a dictionary with the OpenAPI Schema, as illustrated below.
+*'component a'* expects an input with Integer type and emits three outputs with type GCSPath, customized_type and GCRPath. 
+Among these tyeps, Integer, GCSPath, and GCRPath are core types that are predefined in the SDK while customized_type is a user-defined
+type.  
+```yaml
+name: component a
+description: component desc
+inputs:
+  - {name: field_l, type: Integer}
+outputs:
+  - {name: field_m, type: {GCSPath: {openapi_schema_validator: {type: string, pattern: "^gs://.*$" } }}}
+  - {name: field_n, type: customized_type}
+  - {name: field_o, type: GCRPath} 
+implementation:
+  container:
+    image: gcr.io/ml-pipeline/component-a
+    command: [python3, /pipelines/component/src/train.py]
+    args: [
+      --field-l, {inputValue: field_l},
+    ]
+    fileOutputs: 
+      field_m: /schema.txt
+      field_n: /feature.txt
+      field_o: /output.txt
+```
+Similarly, when authors write a component with the decorator, types are annotated in the function signature, as shown below.
```

Comment:
Tech writing hint: Change passive voice to active voice:

Change this:
"types are annotated in the function signature"

To this:
"you annotate the types in the function signature"

Active voice makes it clear that the reader must do the thing, rather than expecting some part of the system to do it for them.

Reply:
Done.